src/pybind/mgr/orchestrator_cli/module.py

```diff
@@ -569,50 +570,42 @@ def _service_instance_action(self, action, svc_type, svc_id):
 
     @orchestrator._cli_write_command(
         'orchestrator mgr update',
-        "name=num,type=CephInt,req=true "
-        "name=hosts,type=CephString,n=N,req=false",
+        "name=num,type=CephInt,req=false "
+        "name=hosts,type=CephString,n=N,req=false "
+        "name=label,type=CephString,n=N,req=false",
         'Update the number of manager instances')
-    def _update_mgrs(self, num, hosts=None):
-        hosts = hosts if hosts is not None else []
+    def _update_mgrs(self, num=None, hosts=[], label=None):
 
-        if num <= 0:
-            return HandleCommandResult(-errno.EINVAL,
-                                       stderr="Invalid number of mgrs: require {} > 0".format(num))
+        placement = orchestrator.PlacementSpec(label=label, count=num, nodes=hosts)
+        placement.validate()
 
-        if hosts:
-            try:
-                hosts = [orchestrator.parse_host_specs(host_spec, require_network=False)
-                         for host_spec in hosts]
-            except Exception as e:
-                msg = "Failed to parse host list: '{}': {}".format(hosts, e)
-                return HandleCommandResult(-errno.EINVAL, stderr=msg)
+        spec = orchestrator.StatefulServiceSpec(placement=placement)
 
-        completion = self.update_mgrs(num, hosts)
+        completion = self.update_mgrs(spec)
         self._orchestrator_wait([completion])
         orchestrator.raise_if_exception(completion)
         return HandleCommandResult(stdout=completion.result_str())
 
     @orchestrator._cli_write_command(
         'orchestrator mon update',
-        "name=num,type=CephInt,req=true "
-        "name=hosts,type=CephString,n=N,req=false",
+        "name=num,type=CephInt,req=false "
+        "name=hosts,type=CephString,n=N,req=false "
+        "name=label,type=CephString,n=N,req=false",
         'Update the number of monitor instances')
-    def _update_mons(self, num, hosts=None):
-        hosts = hosts if hosts is not None else []
+    def _update_mons(self, num=None, hosts=[], label=None):
 
-        if num <= 0:
-            return HandleCommandResult(-errno.EINVAL,
-                                       stderr="Invalid number of mons: require {} > 0".format(num))
+        placement = orchestrator.PlacementSpec(label=label, count=num, nodes=hosts)
+        if not hosts and not label:
+            # Improve Error message. Point to parse_host_spec examples
+            raise Exception("Mons need a host spec. (host, network, name(opt))")
+            # TODO: Scaling without a HostSpec doesn't work right now.
+            # we need network autodetection for that.
```

Comment:
FWIW I think this will work if you ``ceph config set mon public_network 1.2.3.0/24`` or similar

Reply:
Yep, that'd work. Let's move that to a separate PR, though.